### ISSUE TITLE
[apex] Unused Local Variable fails on String query

### ISSUE THREAD

**Issue by tprouvot** [pmd/pmd#3489]
**Affects PMD Version: ** 
- 6.36.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
 [UnusedLocalVariable](https://pmd.github.io/latest/pmd_rules_apex_bestpractices.html#unusedlocalvariable)

**Description:**
When using a variable in a String to build a query, the variable is considered as unused
**Code Sample demonstrating the issue:**

```
String customValue  = 'Test';
		String query = 'SELECT Id FROM Case ';
		query += 'WHERE CustomField__c = :customValue ';
		return Database.getQueryLocator(query);
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI]*
